plugins/services/src/js/containers/services/ServicesContainer.js

```diff
@@ -432,28 +432,54 @@ class ServicesContainer extends React.Component {
     };
   }
 
-  getModals(service) {
-    const modalProps = Object.assign({}, this.state.modal);
+  /**
+   * This function cleans and updates the modalProps property
+   *
+   * This is needed to refresh the state of the service from the store once the
+   * modal is loaded. In our delete group modal for example we need feedback
+   * of the service while it is being deleted. In other places (after deleting
+   * an app) on the other hand, it is possible that the modal is still open,
+   * DCOSStore returns undefined but we still need the (old) information stored
+   * in modalProps.service - and want to close the modal.
+   *
+   * @param {object} stateService - an object which contains id and a service
+   * @param {ServiceTree} [service] - service information
+   * @returns {object} updated and cleaned up modal information (props)
+   */
+  getUpdatedModalProps(stateService, service) {
+    const modalProps = Object.assign({}, stateService);
 
-    // This is needed to refresh the state of the service from the store once the
-    // modal is loaded. In our delete group modal for example we need feedback
-    // of the service while it is being deleted
-    if (!modalProps.service) {
+    // modalProps.service only is set, when a modal is already open, if we want to
```

Comment:
This method is already pretty confusing, and I don't think having all these comments in the method body helps. I recommend putting all the comments above (and try to shorten a bit).

Reply:
Hm, I will try to clarify them a little bit tomorrow. The issue is that this method basicly does two things: "open the modal" and "close the modal" - but not only one modal, but all of them (`id` describes the modal which should be open right now)